x/hard/keeper/repay.go

```diff
@@ -112,7 +112,7 @@ func (k Keeper) ValidateRepay(ctx sdk.Context, sender, owner sdk.AccAddress, coi
 	}
 
 	senderAcc := k.accountKeeper.GetAccount(ctx, sender)
-	senderCoins := senderAcc.SpendableCoins(ctx.BlockTime())
+	senderCoins := k.bankKeeper.SpendableCoins(ctx, senderAcc.GetAddress())
```

Comment:
nit: looks like `senderAcc` is never used so the `GetAccount` line could be removed.